src/modules/Dropdown/Dropdown.js

```diff
@@ -1071,7 +1073,7 @@ export default class Dropdown extends Component {
       <select type='hidden' aria-hidden='true' name={name} value={value} multiple={multiple}>
         <option value='' />
         {_.map(options, (option, i) => (
-          <option key={option.key || option.value} value={option.value}>{option.text}</option>
+          <option key={getKeyOrValue(option.key, option.value)} value={option.value}>{option.text}</option>
```

Comment:
As described in #1720, `0` will break the previous condition while it's a valid key.